src/llmq/quorums_chainlocks.cpp

```diff
@@ -417,6 +416,7 @@ void CChainLocksHandler::ProcessNewChainLock(const NodeId from, CChainLockSig& c
                         fSPV = pnode->pfilter != nullptr;
                     }
                     if (pnode->nVersion >= MULTI_QUORUM_CHAINLOCKS_VERSION && !fSPV && pnode->CanRelay()) {
+                        CInv clsigInv(MSG_CLSIGMQ, ::SerializeHash(clsig));
```

Comment:
This should go outside of the loop, you are needlessly hashing clsig here for every node